src/app/services/component-communication.service.ts

```diff
@@ -13,14 +13,18 @@ export class ComponentCommunicationService {
 
     // used in the listening component.
     // i.e. this.componentCommunicationService = this._componentCommunicationService.on(Events.loginSuccess, () => doSomething());
-    on(event: Events, action: (value: any) => void): Subscription {
+    on(event: Events, action: (value: any) => void, loading?: () => void): Subscription {
         return this._subject$
             .pipe(
                 // filter down based on event name to any events that are emitted out of the subject from the emit method below.
                 filter((e: CpeEvent, i: any) => e.name === event),
-                map((e: CpeEvent) => e.value)
-            )
-            .subscribe(action); // subscribe to the subject to get the data.
+            ).subscribe((e: CpeEvent) => {
```

Comment:
I like this! Can you also update the comment to match?

Reply:
Done in https://github.com/dasch-swiss/dsp-cpe/pull/11/commits/ce69a805ba4a8febc3b83d1ebc2d204ee1071542